### ISSUE TITLE
Script does not preserve pg_hba.conf

### ISSUE THREAD

**Issue by AnthonyMastrean** [bwbroersma/docker-postgres-upgrade#1]
Say I'm upgrading from postgres:9.6 to postgres:10. I started with an empty data dir in 9.6, so the base image's entrypoint setup a pg_hba.conf with password auth for external connections.

https://github.com/docker-library/postgres/blob/0d0485cb02e526f5a240b7740b46c35404aaf13f/9.6/docker-entrypoint.sh#L268

I'm now upgrading to 10 using your script. But, I find the pg_hba.conf reset to defaults (I see the following error and find no md5 allowance in the conf).

```
FATAL:  
no pg_hba.conf entry for host "172.18.0.3", user "postgres", database "postgres", SSL off
```

Your script should have copied my original pg_hba.conf from the old data dir to the new data dir. But, I think the initdb calls resets it? Or you're not copying it, I'm not really sure...

Anyway, the data dir is never empty under 10, so the entrypoint doesn't modify it.

**Comment by AnthonyMastrean** [comment]
I already bring my own postgresql.conf. It doesn't hurt (much) to bring the pg_hba.conf, too. This seems to do the trick.

```dockerfile
COPY pg_hba.conf /etc/postgresql/

...

CMD ["postgres", "-c", "config_file=/etc/postgresql/postgresql.conf", "-c", "hba_file=/etc/postgresql/pg_hba.conf"]
```

**Comment by bwbroersma** [comment]
I will look into it, it's really a Proof of Concept now and some things need to be improved, this is the result of running the [`pg_upgrade`](https://www.postgresql.org/docs/current/pgupgrade.html), which only converts the data and will initializes a new `postgresql.conf` and `pg_hba.conf`.

**Comment by AnthonyMastrean** [comment]
@bwbroersma of course! I was following your comment from this issue https://github.com/docker-library/postgres/issues/37#issuecomment-541321233. I had to adapt the script for the debian version of the postgres base image, but it more or less works.

I think the best future for this script is getting it integrated into the base image, but it doesn't seem like there's much talk of that over there 😢
